src/Tribe/Dependency.php

```diff
@@ -4,7 +4,7 @@
 
```

Comment:
The more I look at this class, I wonder if we can leverage it for the notices in each plugin that we have custom `admin_notice` hooks for and all the extra custom code.

Reply:
Like in https://github.com/moderntribe/tribe-common/blob/acace8801038dc7fbfd7c8a6c3f5de5ca6b3da31/src/Tribe/Dependency.php#L319

We could let the `$version` support the same data that sets `$dependent_plugin` a few lines below. If plugin not registered, it could be added to the list of plugins right there automatically.

Reply:
Are you saying to process just `$plugin` instead of the dependencies and therefore avoid 3 separate calls to `$this->has_valid_dependencies()`?